webdriver/tests/bidi/browsing_context/fragment_navigated/fragment_navigated.py

```diff
@@ -125,14 +125,40 @@ async def test_url_with_base_tag(bidi_session, subscribe_events, inline, new_tab
     )
 
 
-@pytest.mark.parametrize(
-    "context_kind",
-    ["top-level", "iframe"],
-    ids=[
-        "in a top-level context",
-        "in an iframe context",
-    ]
-)
+async def test_iframe(
+    bidi_session, new_tab, url, inline, subscribe_events, wait_for_event
+):
+    initial_url = url(EMPTY_PAGE + '#foo')
```

Comment:
nit: other tests navigate from no hash to `#foo` so this test is a bit different, but would be fine.

Thanks for splitting it off! It saves us quite some time in this test.

Reply:
yeah, I am not sure where the balance that we want to strike lies (I have reduced this test under assumption that the navigation to fragment would be similarly handled in iframes vs top-level). We could, of course, change it to include all hashes (and all ways to navigate?) but in practice there could be many more edge cases which are far more likely (like missing events from cross-origin iframes, fenced frames etc). Not sure how far we should go, so defer to your guidance here.

Reply:
No we most likely wont have to check all the variants here. I only wanted to mention that this test is unique given that is uses a different set of urls. But nothing to worry and we can close this issue if you are fine.